Objects/moduleobject.c

```diff
@@ -679,12 +679,24 @@ module_repr(PyModuleObject *m)
 static PyObject*
 module_getattro(PyModuleObject *m, PyObject *name)
 {
-    PyObject *attr, *mod_name;
+    PyObject *attr, *mod_name, *getattr;
+    PyObject* stack[1] = {name};
     attr = PyObject_GenericGetAttr((PyObject *)m, name);
-    if (attr || !PyErr_ExceptionMatches(PyExc_AttributeError))
+    if (attr || !PyErr_ExceptionMatches(PyExc_AttributeError)) {
         return attr;
+    }
     PyErr_Clear();
     if (m->md_dict) {
+        _Py_IDENTIFIER(__getattr__);
+        getattr = _PyDict_GetItemId(m->md_dict, &PyId___getattr__);
+        if (getattr) {
+            if (!PyCallable_Check(getattr)) {
```

Comment:
There is no a special check for callability of a regular ``__getattr__`` (as well as other special methods).
```
>>> class A:
...     __getattr__ = "Surprise!"
... 
>>> A().x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object is not callable
>>> A()[1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object is not callable
>>> A() + 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object is not callable
```
Of course special checks can be added in all these cases. This would help recognition marginal mistakes. But this will increase the size of the code and can slow down it (because optimizing this code can worsen the optimization of the normal path) and decrease readability. I would defer this change to a separate issue if it is worth to do it at all.